pokemongo_bot/walkers/polyline_generator.py

```diff
@@ -59,19 +64,23 @@ def cached_polyline(origin, destination, google_map_api_key=None):
 
 
 class Polyline(object):
-    def __init__(self, origin, destination, google_map_api_key=None):
+    def __init__(self, origin, destination, google_map_api_key=None, mode="walking"):
         self.origin = origin
         self.destination = tuple(destination)
-        self.DIRECTIONS_API_URL = 'https://maps.googleapis.com/maps/api/directions/json?mode=walking'
-        self.DIRECTIONS_URL = '{}&origin={}&destination={}'.format(self.DIRECTIONS_API_URL,
+        self.DIRECTIONS_API_URL = 'https://maps.googleapis.com/maps/api/directions/json'
+        self.DIRECTIONS_URL = '{}?mode={}&origin={}&destination={}'.format(self.DIRECTIONS_API_URL,
+                mode,
                 '{},{}'.format(*self.origin),
                 '{},{}'.format(*self.destination))
         if google_map_api_key:
             self.DIRECTIONS_URL = '{}&key={}'.format(self.DIRECTIONS_URL, google_map_api_key)
         self._directions_response = requests.get(self.DIRECTIONS_URL).json()
+
         try:
             self._directions_encoded_points = [x['polyline']['points'] for x in
                                                self._directions_response['routes'][0]['legs'][0]['steps']]
+            self._directions_speeds = [x['distance']['value'] / float(x['duration']['value']) for x in
```

Comment:
You're right, that was dumb. Let's see though if we agree that driving should be implemented like flying (sleep and teleport), in which case this code is going out anyway